nixos/modules/virtualisation/containers.nix

```diff
@@ -103,8 +127,87 @@ in
               default = null;
               example = "10.231.136.2";
               description = ''
-                The IPv4 address assigned to <literal>eth0</literal>
-                in the container.
+                The IPv4 address assigned to the container-side of a veth pair
+                (<literal>eth0</literal> in the container).
+                Setting <option>hostAddress</option> or <option>localAddress</option>
+                will create a veth pair with one side in the container appearing as 
+                <literal>eth0</literal>, and the other side in the host as 
+                <literal>ve-<replaceable>container-name</replaceable></literal>.
+              '';
+            };
+
+            macvlanInterface = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              example = "enp1s1";
+              description = ''
+                When this option is set an <literal>eth1</literal> interface
+                will be available within the container that bridges to the host's
+                physical network using macvlan.
+                Note: while macvlan interfaces allow your containers to be accessable 
+                via the the same physical network as the specified host interface, you 
+                may not be able to communicate between the host itself and container.
+              '';
+            };
+      
+            macvlanAddress = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              example = "10.231.136.2";
+              description = ''
+                The IPv4 address assigned to <literal>eth1</literal>
+                (the macvlan interface) in the container.
+              '';
+            };
+
+            macvlanPrefixLength = mkOption {
+              type = types.nullOr types.int;
+              default = 32;
+              example = 16;
+              description = ''
+                The network prefix length for the macvlan interface.
+              '';
+            };
+
+            wantedBy = mkOption {
+              type = types.listOf types.str;
+              default = [ "multi-user.target" ];
+              description = ''
+                List of systemd units/targets that should cause this
+                container to start. Set to <literal>[]</literal> if
+                you do not want this container to start.
+              '';
+            };
+
+            linkJournal = mkOption {
+              type = types.enum [ "auto" "host" "guest" "no" ];
+              default = "auto";
+              description = ''
```

Comment:
Seems the first line of this description is space indented (good) while the remaining lines are tab indented (bad).